sinantalhakosar/backstage
# Complete development of search the information in markdown files on GH

### Description

This PR implements search the information in markdown files. 
Please read;
- README.md: https://github.com/sinantalhakosar/sinan-backstage/blob/development/README.md
- MYREFLECTIONS.md: https://github.com/sinantalhakosar/sinan-backstage/blob/development/MYREFLECTIONS.md

### How to test

Define the list of scanned repositories in app-config.yml file in format:
```yaml
backend:
    search:
        github:
            sources:
                - owner: backstage
                  repo: community-plugins
```

> [!IMPORTANT]
> If you want to add new repository to config, be aware that you need to rerun the application

> [!IMPORTANT]
> Inorder to run frontend side properly on locally, please make sure that you dont have any `/node_modules` folder under `plugins/github-md-docs`. If exists, you can remove it.

1. Run the app
2. Click search from sidebar
3. You can search files by `File name`, `Repository name` and `Repository owner name` 
4. Also click `View Full Results` in modal to search in Search Page aswell.
5. If you click any file name (in modal or page), it should redirect you to internal docs page, where you can: Read basic info and content of file
6. All links (with OpenInNew icon, should redirect you to Github in new tab)
7. Try to break the url after you land `/github-md-docs/<repository_owner>/<repository_name>?path=<mdFilePath>` (with missing parameter or with wrong parameters)
8. It should show EmptyState or Error panel
